### PR TITLE
add player#showDeathScreen method

### PR DESCRIPTION
**Pull request by chmodsayshello**
This simple pull request adds the functionality to open the "You died!" screen via the bukkit api...

### CODE REVIEW

**Comment by chmodsayshello** [comment]
> Your IDE seems to have included a wildcard import resulting in a large diff. If this could be fixed please…

I just noticed that as well, I personally use IntelliJ, I guess it did that because the CraftPlayer class uses all classes within the `net.minecraft.network.protocol.game` package (just an assumption!), but I'll look into it

**Comment by chmodsayshello** [comment]
> > Your IDE seems to have included a wildcard import resulting in a large diff. If this could be fixed please…
> 
> I just noticed that as well, I personally use IntelliJ, I guess it did that because the CraftPlayer class uses all classes within the `net.minecraft.network.protocol.game` package (just an assumption!), but I'll look into it

but even **_IF_** thats the case, might be better to just import all those classes manually (like you suggested)

File: patches/api/0419-add-player-showRespawnScreen-method.patch [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+@@ -3000,4 +3000,15 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
+     @Override
+     Spigot spigot();
+     // Spigot end
++
++    //Paper start
++    /**
++     * Show the death screen to the player
++     * This will NOT actually kill the player, nor increase statistics
++     *<b>Note:</b> Player needs to press respawn more than once to close this screen, since the server will ignore the respawn packet
```
Comment (Doc94) [inline patches/api/0419-add-player-showRespawnScreen-method.patch:20]: space missing?

File: patches/server/0969-add-player-showRespawnScreen-method.patch [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+index 1bada55af5d16437da4d16f9ded55f88a6121eb4..96105e77588fb9345d535c69091207a436aacbb9 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+@@ -38,35 +38,7 @@ import net.minecraft.nbt.CompoundTag;
+ import net.minecraft.network.FriendlyByteBuf;
+ import net.minecraft.network.chat.Component;
+ import net.minecraft.network.chat.PlayerChatMessage;
++import net.minecraft.network.protocol.game.ClientboundPlayerCombatKillPacket;
```
Comment (Doc94) [inline patches/server/0969-add-player-showRespawnScreen-method.patch:15]: you need add the // Paper
or not use the import and use new net.minecraft.network.protocol.game.ClientboundPlayerCombatKillPacket for avoid this.

File: patches/server/0969-add-player-showRespawnScreen-method.patch [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ import net.minecraft.server.level.ChunkMap;
+@@ -3148,4 +3120,14 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
+         return this.spigot;
+     }
+     // Spigot end
++
++    //Paper start
++    @Override
++    public void showDeathScreen(String message){
++        this.getHandle().connection.send(new ClientboundPlayerCombatKillPacket(this.getEntityId(),-1,Component.nullToEmpty(message)));
```
Comment (Doc94) [inline patches/server/0969-add-player-showRespawnScreen-method.patch:27]: missing space after ","

File: patches/server/0969-add-player-showRespawnScreen-method.patch [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
++        /*the second argument (hardcoded integer), specifies what "killed" the player, -1 means minecraft doesn't know...
++        https://wiki.vg/Protocol#Combat_Death
++        */
```
Comment (Doc94) [inline patches/server/0969-add-player-showRespawnScreen-method.patch:30]: this can be added in the description of the patch rather add here?

**Review by jpenilla — CHANGES_REQUESTED** [review]
Not a fan of adding API that misuses the protocol putting the client in an invalid state where you need to click a button twice to exit the screen. Should just use userdev/plib if you want to mess with packets like this.

**Comment by kashike** [comment]
I am going to go ahead and close this pull request for the reason @jpenilla stated above.

**Pull request closed without merging**